cypress/support/commands.ts

```diff
@@ -15,42 +16,58 @@ export const signin = (): Cypress.Chainable<Cypress.Response> => {
-  return cy
-    .setupUser()
-    .then(resp => {
-      Object.entries(resp.body).forEach(([k, v]) => {
-        cy.log(k, JSON.stringify(v))
+  return cy.setupUser().then(response => {
+    // const route = Cypress.env('inkind') ? '/' : '/api/v2/signin'
+    cy.visit('/')
+      .then(() => {
+        const username = Cypress._.get(
+          response,
+          'body.user.name',
+          Cypress.env('username')
+        )
+        cy.get('#login').type(username)
       })
-      // cy.log('user setup response', JSON.stringify(resp, null, 2))
-      cy.visit('/api/v2/signin')
-    })
-    .then(() => cy.get('#login').type(Cypress.env('username')))
-    .then(() => cy.get('#password').type(Cypress.env('password')))
-    .then(() => cy.get('#submit-login').click())
-    .then(() => {
-      cy.get('body').then($body => {
-        /**
-         * we are conditionally rendering this test case since it's only
-         * relevant to CLOUD tests in order to click the `Grant Access` button
-         * that's rendered by Dex in the CLOUD development environment.
-         *
-         * We are using this conditional test based on the following doc suggestions:
-         * https://docs.cypress.io/guides/core-concepts/conditional-testing.html#Element-existence
-         **/
-        if ($body.find('.theme-btn--success').length) {
-          cy.get('.theme-btn--success').click()
-        }
+      .then(() => cy.get('#password').type(Cypress.env('password')))
+      .then(() => cy.get('#submit-login').click())
+      .then(() => {
+        cy.get('body').then($body => {
+          /**
+           * we are conditionally rendering this test case since it's only
+           * relevant to CLOUD tests in order to click the `Grant Access` button
+           * that's rendered by Dex in the CLOUD development environment.
+           *
+           * We are using this conditional test based on the following doc suggestions:
+           * https://docs.cypress.io/guides/core-concepts/conditional-testing.html#Element-existence
+           **/
+          if ($body.find('.theme-btn--success').length) {
+            cy.get('.theme-btn--success').click()
+          }
+        })
       })
+      .then(() => cy.location('pathname').should('not.eq', '/signin'))
+      .then(() => cy.wrapOrgAndBucket())
+  })
+}
+
+export const wrapOrgAndBucket = (): Cypress.Chainable<Cypress.Response> => {
+  return cy
+    .request({
+      method: 'GET',
+      url: '/api/v2/orgs',
     })
-    .then(() => cy.location('pathname').should('not.eq', '/signin'))
-    .then(() =>
+    .then(orgsResponse => {
+      const org = orgsResponse.body.orgs[0]
+      cy.wrap(org).as('org')
+
       cy.request({
         method: 'GET',
-        url: '/api/v2/orgs',
+        url: '/api/v2/buckets',
+        qs: {orgID: org.id},
+      }).then(bucketsResponse => {
+        const buckets = bucketsResponse.body.buckets as Array<Bucket>
+        const bucket = buckets.find(b => b.name === 'defbuck')
```

Comment:
think the logic here is "find the first bucket who has an org id that matches my own (to get around sample data buckets) and not _tasks or _monitoring

Reply:
ok